### ISSUE TITLE
global_step_tensor value

### ISSUE THREAD

**Issue by Auth0rM0rgan** [vonclites/checkmate#1]
Hey @vonclites, Thanks for sharing your code. 

I am trying to use your code inside of my model to save the best checkpoint according to my validation accuracy but getting an error related to `global_step_tensor` value. I am curious what should be the value of `global_step_tensor` in below line?

`best_ckpt_saver.handle(accuracy, sess, global_step_tensor)`

**Comment by vonclites** [comment]
This would be the tensor variable you get when calling 
`tf.train.get_or_create_global_step()`

**Comment by arminXerror** [comment]
Hey @vonclites 

Would you explain more precisely how your code is working? because It's not working for me

Before starting `tf.Session()`, I add these line as you mentioned in your repo


`best_ckpt_saver = BestCheckpointSaver(save_dir=checkpoint_dir, num_to_keep=3, maximize=True)`


Inside of my `tf.Session()` and calculating my validation accuracy and everything, I added this line

`best_ckpt_saver.handle(val_acc, sess, global_step_tensor=tf.train.get_or_create_global_step())
`
to save the checkpoint with best validation accuracy.

The code is working but is not saving the best one. I mean, it's overwriting the checkpoint file (work same as 
`saver = tf.train.Saver()`) and also my global _step is always 0 and as I set num_to_keep=3, It should save 3 checkpoints, isn't it? any idea why I am not able to save the best one?

Thanks in advance.

**Comment by Auth0rM0rgan** [comment]
@vonclites, Thanks for the help I got global_step value in another way

First, I have assigned global_step to a `tf.Variable(0, name='global_step', trainable=False)`  

then I fed it to my optimizer as follow:
`apply_grads = optimizer.apply_gradients(grads_and_vars=grads,global_step=global_steps)`

in the end, I assigned like this 
`best_ckpt_saver.handle(val_acc, sess, global_step_tensor=global_steps)`

You can close this issue.
